server/src/main/java/io/druid/server/coordination/ZkCoordinator.java

```diff
@@ -252,28 +257,53 @@ public void loadLocalCache()
   {
     final long start = System.currentTimeMillis();
     File baseDir = config.getInfoDir();
-    if (!baseDir.exists() && !config.getInfoDir().mkdirs()) {
-      return;
+
+    //Move files to new location. This will be removed after 0.11 has released
+    File[] segmentsToMove = baseDir.listFiles();
+    for (File cacheFile : segmentsToMove) {
+      try {
+        final DataSegment segment = jsonMapper.readValue(cacheFile, DataSegment.class);
+        File toLocation = segmentManager.getSegmentLocation(segment);
+        toLocation.mkdirs();
+        Files.move(cacheFile.toPath(),
+            new File(toLocation, "info.json").toPath(),
+            StandardCopyOption.REPLACE_EXISTING);
+      } catch (IOException e) {
+        log.makeAlert(e, "Failed to move segmentInfo file")
+            .addData("file", cacheFile)
+            .emit();
+      } catch (Exception e) {
+        log.makeAlert(e, "Failed to load segment from segmentInfo file")
+            .addData("file", cacheFile)
+            .emit();
+      }
     }
 
     List<DataSegment> cachedSegments = Lists.newArrayList();
-    File[] segmentsToLoad = baseDir.listFiles();
-    int ignored = 0;
-    for (int i = 0; i < segmentsToLoad.length; i++) {
-      File file = segmentsToLoad[i];
-      log.info("Loading segment cache file [%d/%d][%s].", i, segmentsToLoad.length, file);
+    List<File> segmentsToLoad = Collections.EMPTY_LIST;
+    for (Path location : segmentManager.getSegmentLocations()) {
       try {
-        final DataSegment segment = jsonMapper.readValue(file, DataSegment.class);
+        segmentsToLoad = Files.walk(location, 5)
```

Comment:
It was 5 because of the current structure of the files and how they are nested. Will change it to infinite.